### PR TITLE
Add gradle ktfmt spotless config to all projects

### LINKED ISSUE #3435: Autoformat/enforce formatting of build scripts (kts)
https://github.com/open-telemetry/opentelemetry-java/pull/3427 showed that the build scripts were not consistently formatted. To avoid such large (or any!) formatting-only PRs in the future, we should have something like spotless+googleJavaFormat for the kts files.

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #3435. 

Uses [ktfmt](https://github.com/facebookincubator/ktfmt) to apply `google-java-format` style to all `*.gradle.kts` build files.

Opening as a draft to gauge interest in the approach and the resultant style. 

See [kotlin spotless config](https://github.com/diffplug/spotless/tree/main/plugin-gradle#kotlin) docs for additional configuration options.

### CODE REVIEW

File: all/build.gradle.kts [hunk 1]
```diff
@@ -1,24 +1,19 @@
-plugins {
-  id("otel.java-conventions")
-}
+plugins { id("otel.java-conventions") }
```
Comment (anuraaga) [inline all/build.gradle.kts:1]: Don't think Gradle DSL looks good with all the one-liners (which would look good for normal lambda invocations in general, but not DSL)

File: all/build.gradle.kts [hunk 2]
```diff
@@ -66,21 +59,25 @@ tasks.named<JacocoReport>("jacocoTestReport") {
...
+          !it.absolutePath.contains("AutoValue_") &&
+          // TODO(anuraaga): Remove exclusion after enabling coverage for jfr-events
+          !it.absolutePath.contains("io/opentelemetry/sdk/extension/jfr")
+      }
+    )
   }
   additionalSourceDirs(sourcesPath.incoming.artifactView { lenient(true) }.files)
-  executionData(coverageDataPath.incoming.artifactView { lenient(true) }.files.filter { it.exists() })
+  executionData(
+    coverageDataPath.incoming.artifactView { lenient(true) }.files.filter { it.exists() }
```
Comment (anuraaga) [inline all/build.gradle.kts:79]: This change looks reasonable

File: api/all/build.gradle.kts [hunk 3]
```diff
@@ -7,7 +7,9 @@ plugins {
 }
 
 description = "OpenTelemetry API"
+
```
Comment (anuraaga) [inline api/all/build.gradle.kts:10]: Don't think these newlines improve (wonder why ktfmt would go this far)

File: bom-alpha/build.gradle.kts [hunk 4]
```diff
@@ -16,15 +18,12 @@ rootProject.subprojects.forEach { subproject ->
 afterEvaluate {
   dependencies {
     constraints {
-      rootProject.subprojects
+      rootProject
+        .subprojects
```
Comment (anuraaga) [inline bom-alpha/build.gradle.kts:22]: This change looks like a good one

File: build.gradle.kts [hunk 5]
```diff
@@ -9,28 +9,25 @@ plugins {
 }
 
 if (!JavaVersion.current().isJava11Compatible()) {
-  throw GradleException("JDK 11 or higher is required to build. " +
-    "One option is to download it from https://adoptopenjdk.net/. If you believe you already " +
-    "have it, please check that the JAVA_HOME environment variable is pointing at the " +
-    "JDK 11 installation.")
+  throw GradleException(
+    "JDK 11 or higher is required to build. " +
```
Comment (anuraaga) [inline build.gradle.kts:13]: This change looks good

**Review by anuraaga — COMMENTED** [review]
Thanks for taking a look at this @jack-berg - definitely interesting to try out formatters. I think there are significant gaps here so wouldn't want to go with this formatter unfortunately.

I notice that ktlint mentions `*.kts` files on its README and it's not mentioned on ktfmt so perhaps it has more idiomatic formatting of kts files - do you mind giving it a try?

**Comment by anuraaga** [comment]
Realized ktlint is what I'm using in my distro :)

https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/build.gradle.kts#L71

Formats well, except for some corner cases like having to exclude settings.gradle

File: all/build.gradle.kts (hunk 1 above)
Comment (Oberon00) [reply all/build.gradle.kts:1]: I have no issue with that. I would much prefer a general kts/Kotlin formatter over no formatter, even if it sometimes leads to sub-optimal results.

**Comment by jack-berg** [comment]
@anuraaga I opened an alternative [draft PR](https://github.com/open-telemetry/opentelemetry-java/pull/3543) using ktlint instead of ktfmt. It seems less opinionated, including allowing short closures to be on multiple lines. Personally I'm partial to closures that are not inlined, so I prefer ktlint as well.

**Comment by anuraaga** [comment]
Closing since we used ktlint - thanks!

**Pull request closed without merging**